$OpenBSD: patch-configure_ac,v 1.1 2013/04/01 14:30:32 brad Exp $

Remove incredibly stupid autoconf test which assumes that the lack
of the getaddrinfo() flag AI_ADDRCONFIG means getaddrinfo() is not
thread-safe. Although that was the case in the past it is no longer
true.

--- configure.ac.orig	Mon Apr  1 03:13:31 2013
+++ configure.ac	Mon Apr  1 03:24:30 2013
@@ -219,29 +219,7 @@ AC_SEARCH_LIBS([dlopen], [dl], [],
 DNS_TEXT=blocking
 if test "x$TDNS" != "xno"; then
 	old_TDNS=$TDNS
-	AX_PTHREAD([
-		AC_MSG_CHECKING([whether getaddrinfo() supports AI_ADDRCONFIG])
-		AC_COMPILE_IFELSE([
-			AC_LANG_PROGRAM([[
-				#include <sys/types.h>
-				#include <sys/socket.h>
-				#include <netdb.h>
-			]], [[
-				int x = AI_ADDRCONFIG;
-				(void) x;
-			]])
-		], [
-			AC_MSG_RESULT([yes])
-			TDNS=yes
-		], [
-			AC_MSG_RESULT([no])
-			TDNS=no
-			# Note that broken systems, such as OpenBSD, NetBSD, which don't support AI_ADDRCONFIG,
-			# also have thread-unsafe getaddrinfo().
-		])
-	], [
-		TDNS=no
-	])
+	AX_PTHREAD([TDNS=yes], [TDNS=no])
 	if test "x$TDNS" = "xyes"; then
 		DNS_TEXT=threads
 		AC_DEFINE([HAVE_THREADED_DNS], [1], [Define if threaded DNS is enabled])
